### PR TITLE
[CI] Delete validator pod if stuck during E2E test

### PR DESCRIPTION
**Pull request by okdas**
### Human Summary

Delete stuck validator Pod during E2E test.

### AI Summary

reviewpad:summary

## Issue

DevNets now use a new helm chart that provisions validators as StatefulSets. As a result, if such validator had unsuccessful deployment, it won't be upgraded to the newer version due to StatefulSet uptime guarantees. We need to kill that pod, so StatefulSet controller creates a new one with correct image.

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [x] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

- [ ] **Run all unit tests**: `make go_develop_and_test`
- [ ] **Run E2E tests locally**: `make test_e2e`
- [ ] **Run E2E tests on DevNet**: Add the `devnet-test-e2e` label to the PR. This is VERY expensive, only do it after all the reviews are complete.

## Sanity Checklist

- [ ] I have tested my changes using the available tooling
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, updated documentation and left TODOs throughout the codebase

### CODE REVIEW

File: .github/workflows-helpers/run-e2e-test.sh [hunk 1]
```diff
@@ -1,19 +1,28 @@
...
+echo "Checking for ready validator pod with image SHA ${IMAGE_TAG} or pods needing recreation..."
 while :; do
     # Get all pods with the matching purpose
     PODS_JSON=$(kubectl get pods -n ${NAMESPACE} -l pokt.network/purpose=validator -o json)
 
     # Check if any pods are running and have the correct image SHA
     READY_POD=$(echo $PODS_JSON | jq -r ".items[] | select(.status.phase == \"Running\") | select(.spec.containers[].image | contains(\"${IMAGE_TAG}\")) | .metadata.name")
+    
+    # Check for non-running pods with incorrect image SHA to delete
+    INCORRECT_POD=$(echo $PODS_JSON | jq -r ".items[] | select(.status.phase != \"Running\") | select(.spec.containers[].image | contains(\"${IMAGE_TAG}\") | not) | .metadata.name")
```
Comment (Olshansk) [inline .github/workflows-helpers/run-e2e-test.sh:13]: NIT: What do you think of adding an intermediary helper variable called `NON_RUNNING_PODS` so it's easier to read/nuderstand/main?

File: .github/workflows-helpers/run-e2e-test.sh [hunk 2]
```diff
@@ -1,19 +1,28 @@
...
 
     if [[ -n "${READY_POD}" ]]; then
         echo "Ready pod found: ${READY_POD}"
         break
+    elif [[ -n "${INCORRECT_POD}" ]]; then
+        echo "Non-ready pod with incorrect image found: ${INCORRECT_POD}. Deleting..."
+        kubectl delete pod -n ${NAMESPACE} ${INCORRECT_POD}
+        echo "Pod deleted. StatefulSet will recreate the pod."
+        # You may want to wait a bit here before the next iteration
```
Comment (Olshansk) [inline .github/workflows-helpers/run-e2e-test.sh:22]: Can you update the comment explainin WHY we want to wait?

File: .github/workflows-helpers/run-e2e-test.sh [hunk 3]
```diff
@@ -1,19 +1,28 @@
...
+        sleep 10
     else
-        echo "Validator with with an image ${IMAGE_TAG} is not ready yet. Will retry in 10 seconds..."
+        echo "Validator with image ${IMAGE_TAG} is not ready yet and no incorrect pods found. Will retry in 10 seconds..."
```
Comment (Olshansk) [inline .github/workflows-helpers/run-e2e-test.sh:25]: Please update message. What are we retrying in 10 seconds?

**Review by Olshansk — CHANGES_REQUESTED** [review]

File: .github/workflows-helpers/run-e2e-test.sh (hunk 2 above)
Comment (okdas) [reply .github/workflows-helpers/run-e2e-test.sh]: @Olshansk oh, I updated comment. Thanks!

**Review by Olshansk — APPROVED** [review]

**Pull request merged**